pkg/util/kernel/proc.go

```diff
@@ -107,3 +108,91 @@ func GetProcessEnvVariable(pid int, procRoot string, envVar string) (string, err
 
 	return getEnvVariableFromBuffer(envFile, envVar), nil
 }
+
+// ErrMemFdFileNotFound is an error for when there was no file
+// found for a given process.
+var ErrMemFdFileNotFound = errors.New("memfd file not found")
+
+// GetProcessMemFdFile reads a maximum amount of bytes from the
+// memFdFileName if it was found.
+func GetProcessMemFdFile(pid int, procRoot string, memFdFileName string, memFdMaxSize int) ([]byte, error) {
+	path, found := findMemFdFilePath(pid, procRoot, memFdFileName)
+	if !found {
+		return nil, ErrMemFdFileNotFound
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(io.LimitReader(file, int64(memFdMaxSize)))
+	if err != nil {
+		return nil, err
+	}
+
+	if len(data) == memFdMaxSize {
```

Comment:
Because of this, I don't think you can ever read exactly `memFdMaxSize` without error. There is no way to differentiate between the `LimitReader` limiting you and reading the exact max.

Reply:
Hey @brycekahle! I'm going to attempt to shephard this PR along while Stan is out on PTO. I pushed a solution where we read max+1 bytes to compare against, let me know what you think.